Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -554,6 +561,19 @@ extension SourceKitServer {
     }
   }
 
+  func didChangeConfiguration(_ notification: Notification<DidChangeConfigurationNotification>, workspace: Workspace) {
+    switch notification.params.settings {
+    case .sourcekitlsp(let settings):
+      if let indexVisibility = settings.indexVisibility {
+        queue.async {
```

Comment:
Isn't `didChangeConfiguration` a notification handler?  If so, it's already on `queue`.